packages/protocol/contracts/governance/Governance.sol

```diff
@@ -1066,11 +1066,18 @@ contract Governance is
     // solhint-disable-next-line not-rely-on-time
     if (now >= lastDequeue.add(dequeueFrequency)) {
       uint256 numProposalsToDequeue = Math.min(concurrentProposals, queue.list.numElements);
-      uint256[] memory dequeuedIds = queue.popN(numProposalsToDequeue);
+      (uint256[] memory keys, ) = queue.getElements();
```

Comment:
We definitely don't want to do this. This opens a DoS vulnerability, where an attacker creates a bunch of proposals, making the queue so long that reading it into memory runs out of gas. What was the reason you wanted to change this?

Something to always keep in mind when working with arrays in Solidity, btw. Reading a whole array into memory is only safe if you are 100% guaranteed the array's length is limited by a small constant (in particular, it's not dependent on arbitrary users' actions).

Reply:
As discussed, I reverted this change and updated tests accordingly